sale_order_type/models/sale_order.py

```diff
@@ -35,6 +35,23 @@ def onchange_type_id(self):
             if order.type_id.incoterm_id:
                 order.incoterm = order.type_id.incoterm_id.id
 
+    @api.multi
+    @api.onchange('order_line')
```

Comment:
You should probably use `order_line.product_id`

Reply:
@eLBati fields in `onchange` cannot be dotted